docs/raw/push-notification-server.md

```diff
@@ -318,21 +314,29 @@ message PushNotificationQueryResponse {
 
 Otherwise a response MUST NOT be sent.
 
-Multiple queries MAY be sent for the same token from different public keys to
-increase deniability.
+If `allowed_user_list` is not set `access_token` MUST be set and `encrypted_access_tokens` MUST NOT
+be set.
+
+If `allowed_user_list` is set `encrypted_access_tokens` MUST be set and `access_token` MUST NOT be set.
+
+If `access_token` is returned, this SHOULD be used to send push notifications.
 
-A client MAY send request to the server using an ephemeral key at first, in order not to reveal 
-their identity.
+[//]: TODO: Add more details on the exact encryption method, AES-CTR etc
 
-If the request is not successful, they should assume that some public key filtering is
-in place at the server level and they MAY retry the request using their real identity.
+If `encrypted_access_tokens` is returned, the client SHOULD decrypt each
+token by generating a `AES` symmetric key from the Diffie–Hellman between the
+target client and the querying client.
+If a token is decrypted successfully it SHOULD be used to send push notifications.
```

Comment:
(As discussed) We can add how to identify successful decryption - "If AES decryption succeeds it will return a valid uuid which is what is used for `access_token`."